pydarn/plotting/rtp.py

```diff
@@ -398,17 +398,27 @@ def plot_range_time(cls, dmap_data: List[dict], parameter: str = 'v',
                      'elv': PyDARNColormaps.PYDARN}
             cmap = cmaps[parameter]
 
-        if isinstance(groundscatter, str):
-            cmap.set_under(groundscatter, 1.0)
-        elif groundscatter:
-            cmap.set_under('grey', 1.0)
 
         # set the background color, this needs to happen to avoid
         # the overlapping problem that occurs
         cmap.set_bad(color=background, alpha=1.)
         # plot!
         im = ax.pcolormesh(time_axis, y_axis, z_data, lw=0.01,
                            cmap=cmap, norm=norm, **kwargs)
+
+        if isinstance(groundscatter, str):
+            ground_scatter = np.ma.masked_where( z_data != -1000000, z_data)
+            gs_color = colors.ListedColormap([groundscatter])
+            im2 = ax.pcolormesh(time_axis, y_axis, ground_scatter, lw=0.01,
+                           cmap=gs_color, norm=norm, **kwargs)
```

Comment:
Just want to check this is working as expected. If groundscatter gets given a string instead of a bool character, then it plots ground scatter in grey. From line 411, I would expect that if the string was say 'black' it would plot it in black, but it doesn't, it plots in grey still? Is the grey the expected outcome, and this is kinda a way to catch any user input errors?

And as a side, if you use the string 'False' by accident instead of the bool it plots the gs in grey.

Reply:
So I tried plotting in black and 'False' both give me expected behaviour. 

What is the code you are using?

Reply:
Is the expected behaviour that it should plot out the groundscatter as black then, if `groundscatter='black'`?

Just the code above but with 
`a = pydarn.RTP.plot_summary(fitacf_data, beam_num=5, groundscatter='black')`
and
`a = pydarn.RTP.plot_summary(fitacf_data, beam_num=5, groundscatter='False')`

Both give me GS in grey as if it's testing the truthiness of the two (same as first plot above).
I'm gonna go out on a limb and say it's possibly a matplotlib thing again, so will downgrade matplotlib and test again too.

Reply:
Never mind, it's mysteriously working as expected now!